types/index.d.ts

```diff
@@ -5210,7 +5191,47 @@ declare namespace math {
     /**
      * Determine the type of a variable.
      */
-    typeOf(): MathJsChain;
+    typeOf<TReturn = ReturnType<MathJsStatic['typeOf']>>(): MathJsChain<TReturn>;
+  }
+
+  interface MathJsNumberChain extends MathJsChain<number> {
+    bignumber(): MathJsBigNumberChain;
+    number(): MathJsNumberChain;
+    boolean(): MathJsBooleanChain;
+  }
+
+  interface MathJsStringChain extends MathJsChain<string> {
+    bignumber(): MathJsBigNumberChain;
+    number(): MathJsNumberChain;
+    boolean(): MathJsBooleanChain;
+  }
+
+  interface MathJsFractionChain extends MathJsChain<Fraction> {
+    bignumber(): MathJsBigNumberChain;
+    number(): MathJsNumberChain;
+  }
+
+  interface MathJsNullChain extends MathJsChain<null> {
+    bignumber(): MathJsBigNumberChain;
+    number(): MathJsNumberChain;
+    boolean(): MathJsBooleanChain;
+  }
+
+  interface MathJsBigNumberChain extends MathJsChain<BigNumber> {
+    bignumber(): MathJsBigNumberChain;
+    number(): MathJsNumberChain;
+  }
+
+  interface MathJsBooleanChain extends MathJsChain<boolean> {
+    bignumber(): MathJsBigNumberChain;
+    number(): MathJsNumberChain;
+    boolean(): MathJsBooleanChain;
+  }
+
+  interface MathJsMathCollectionChain extends MathJsChain<MathCollection> {
+    bignumber(): MathJsBigNumberChain;
+    number(): MathJsNumberChain;
+    boolean(): MathJsBooleanChain;
```

Comment:
Separate subtypes of MathJsChain for each chainable function return type with applicable functions defined on each